pkg/table/table.go

```diff
@@ -202,7 +202,7 @@ func (opt *UpdateRecordOpt) SkipWriteUntouchedIndices() bool {
 
 // GetAddRecordOpt creates a AddRecordOpt.
 func (opt *UpdateRecordOpt) GetAddRecordOpt() *AddRecordOpt {
-	return &AddRecordOpt{commonMutateOpt: opt.commonMutateOpt}
+	return &AddRecordOpt{commonMutateOpt: opt.commonMutateOpt, isUpdate: true}
```

Comment:
Is this only needed for partitioned tables, or also for non-partitioned tables, where the PK/_tidb_rowid is updated or replaced (like REPLACE INTO)?

Reply:
non-partitioned table handles it. https://github.com/pingcap/tidb/blob/9ef5b59f6dc0ddac6b1151d4b62368cfca2457ec/pkg/executor/write.go#L184